ui/src/components/Settings/Ombi/index.tsx

```diff
@@ -0,0 +1,226 @@
+import TestButton from '../../Common/TestButton'
+
+const OmbiSettings = () => {
+  const settingsCtx = useContext(SettingsContext)
+  const hostnameRef = useRef<HTMLInputElement>(null)
+  const portRef = useRef<HTMLInputElement>(null)
+  const apiKeyRef = useRef<HTMLInputElement>(null)
+  const [hostname, setHostname] = useState<string>()
+  const [port, setPort] = useState<string>()
+  const [error, setError] = useState<boolean>()
+  const [changed, setChanged] = useState<boolean>()
+  const [testBanner, setTestbanner] = useState<
+    | {
+        status: boolean
+        message: string
+      }
+    | undefined
+  >()
+
+  useEffect(() => {
+    document.title = 'Maintainerr - Settings - Ombi'
+  }, [])
+
+  useEffect(() => {
+    // hostname
+    setHostname(removePortFromUrl(settingsCtx.settings.ombi_url))
+    // @ts-ignore
+    hostnameRef.current = {
+      value: removePortFromUrl(settingsCtx.settings.ombi_url) ?? '',
+    }
+
+    // port
+    setPort(getPortFromUrl(settingsCtx.settings.ombi_url))
+    // @ts-ignore
+    portRef.current = {
+      value: getPortFromUrl(settingsCtx.settings.ombi_url) ?? '',
+    }
+
+    // api key
+    // @ts-ignore
+    apiKeyRef.current = {
+      value: settingsCtx.settings.ombi_api_key ?? '',
+    }
+  }, [settingsCtx])
+
+  const submit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+
+    // if port not specified, but hostname is. Derive the port
+    if (!portRef.current?.value && hostnameRef.current?.value) {
+      const derivedPort = hostnameRef.current.value.includes('http://')
+        ? 80
+        : hostnameRef.current.value.includes('https://')
+          ? 443
+          : 80
+
+      if (derivedPort) {
+        setPort(derivedPort.toString())
+        // @ts-ignore
+        portRef.current = { value: derivedPort.toString() }
+      }
+    }
+
+    if (
+      hostnameRef.current?.value &&
+      apiKeyRef.current?.value &&
+      portRef.current?.value
+    ) {
+      const hostnameVal = hostnameRef.current.value.includes('http://')
+        ? hostnameRef.current.value
+        : hostnameRef.current.value.includes('https://')
+          ? hostnameRef.current.value
+          : portRef.current.value == '443'
+            ? 'https://' + hostnameRef.current.value
+            : 'http://' + hostnameRef.current.value
+
+      const payload = {
+        ombi_url: addPortToUrl(hostnameVal, +portRef.current.value),
+        ombi_api_key: apiKeyRef.current.value,
+      }
+      const resp: { code: 0 | 1; message: string } = await PostApiHandler(
+        '/settings',
+        {
+          ...settingsCtx.settings,
+          ...payload,
+        },
+      )
+      if (Boolean(resp.code)) {
+        settingsCtx.addSettings({
+          ...settingsCtx.settings,
+          ...payload,
+        })
+        setError(false)
+        setChanged(true)
+      } else setError(true)
+    } else {
+      setError(true)
+    }
+  }
```

Comment:
Please refactor this to use React Hook Form. See the Tautulli settings form for an example.